app/models/portfolio.rb

```diff
@@ -22,7 +24,25 @@ def add_portfolio_item(portfolio_item)
   end
 
   def metadata
-    {:user_capabilities => user_capabilities,
-     :shared            => self.access_control_entries.any?}
+    {:user_capabilities => user_capabilities}.tap do |data|
+      data[:statistics] = statistics
+    end
+  end
+
+  def update_statistics
+    current_stats = stats
+    return if current_stats.eql?(statistics)
```

Comment:
This check prevents tags changes that are not approval based and `before_save` calls on this calss from forcing an addition db commit if nothing actually changed.